Apps/WebClient/src/ClientApp/app/store/modules/auth/mutations.ts

```diff
@@ -0,0 +1,31 @@
+import { MutationTree } from 'vuex';
+import { AuthState, Authentication } from '@/models/authState';
+import { stat } from 'fs';
+
+export const mutations: MutationTree<AuthState> = {
+    authenticationRequest(state: AuthState) {
+        state.error = false;
+        state.statusMessage = 'loading';
+        state.requestedRoute = '';
+    },
+    authenticationLoaded(state: AuthState, payload: Authentication) {
+        state.error = false;
+        state.authentication = payload;
+        state.statusMessage = 'success';
+        state.requestedRoute = '';
+    },
+    authenticationError(state: AuthState, errorMessage: string) {
+        state.error = true;
+        state.authentication = undefined;
+        state.statusMessage = errorMessage;
+    },
+    logout(state: AuthState) {
```

Comment:
Looks like the mutations work as a state diagram for the AuthState. What if we also add logoutRequested, logoutComplete as mutations that set the entire state as the logout flow happens?